# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1257142590 -3600
#      Mon Nov 02 07:16:30 2009 +0100
# Node ID 82fe4db20decd3d0ee6e12784695543fe635803f
# Parent  7fc1c8c47b8698809945323f89beb9853259190b
fix bad FLOAT_TRUNCATE in mx-inlines.cc

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2009-11-02  Jaroslav Hajek  <highegg@gmail.com>
+
+	* mx-inlines.cc (twosum_accum): Remove FLOAT_TRUNCATE.
+
 2009-10-28  John W. Eaton  <jwe@octave.org>
 
 	* PermMatrix.cc: Don't include error.h.
 
 2009-10-27  David Bateman  <dbateman@free.fr>
 
 	* SparsedbleLU.cc (SparseLU::SparseLU (const SparseMatrix&,
 	const Matrix&, bool), SparseLU::SparseLU (const SparseMatrix&,
diff --git a/liboctave/mx-inlines.cc b/liboctave/mx-inlines.cc
--- a/liboctave/mx-inlines.cc
+++ b/liboctave/mx-inlines.cc
@@ -1244,17 +1244,17 @@ do_mx_diff_op (const ArrayType& src, int
 // T. Ogita, S. M. Rump, S. Oishi:
 // Accurate Sum And Dot Product,
 // SIAM J. Sci. Computing, Vol. 26, 2005
 
 template <class T>
 inline void twosum_accum (T& s, T& e, 
                           const T& x)
 {
-  FLOAT_TRUNCATE T s1 = s + x, t = s1 - s, e1 = (s - (s1 - t)) + (x - t);
+  T s1 = s + x, t = s1 - s, e1 = (s - (s1 - t)) + (x - t);
   s = s1;
   e += e1;
 }
 
 template <class T>
 inline T
 mx_inline_xsum (const T *v, octave_idx_type n) 
 {
